golint/golint.go

```diff
@@ -69,25 +73,61 @@ func exists(filename string) bool {
 }
 
 func lintFiles(filenames ...string) {
-	files := make(map[string][]byte)
+	files := make(map[string]struct{})
+	pkgs := make(map[string]map[string][]byte)
 	for _, filename := range filenames {
+		if _, ok := files[filename]; ok {
+			continue
+		}
+
 		src, err := ioutil.ReadFile(filename)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
 			continue
 		}
-		files[filename] = src
+		files[filename] = struct{}{}
+
+		pkgName := path.Dir(filename)
+		if strings.HasSuffix(filename, "_test.go") {
```

Comment:
I am revisiting this PR and I thought about your suggestion. I do not think that this makes it particularly simpler. It would also result into the same problem I introduced here, that _test.go files are read in twice. So i think the PR is fine as it is even if it implements parts of go/build.

However, I did some work for https://github.com/kisielk/errcheck which is using go/loader and I think that this would be a better solution. I will try it out and get back to you.